src/SampSharp.GameMode/World/BasePlayer.cs

```diff
@@ -116,7 +116,13 @@ public virtual string Name
                 PlayerInternal.Instance.GetPlayerName(Id, out var name, MaxNameLength);
                 return name;
             }
-            set => PlayerInternal.Instance.SetPlayerName(Id, value);
+            set
+            {
+                if(PlayerInternal.Instance.SetPlayerName(Id, value) == -1)
+                {
+                    throw new Exception("The name is already in use, too long or has invalid characters.");
```

Comment:
Could you change it to `InvalidOperationException`?

Reply:
OK. But wait, I'm adding similar return codes handling on the other setters, seems that all lacks these checks.